cluster-provision/images/vm-image-builder/customize-image.sh

```diff
@@ -11,6 +11,21 @@ function cleanup() {
   virsh undefine "${DOMAIN_NAME}" || true
 }
 
+function wait_for_vm_state() {
+  count=0
+  retries=100
+  until [[ $(virsh list --state-$1 --name |grep . ) =~ $DOMAIN_NAME || $count -gt $retries ]]; do
```

Comment:
To remove the extra line that is generated by the `virsh list --state-shutoff --name`. Here is an example of the same command using the `--state-running`.
```
[root@localhost ~]# virsh list --name --state-running
provision-vm

[root@localhost ~]#
```

With `grep .`, it removes the extra line:
```
[root@localhost ~]# virsh list --name --state-running |grep .
provision-vm
[root@localhost ~]#
```

If you prefer a different way of removing the empty spaces, I'm all ears :smile:

Reply:
I think there is no need to use `[[` + `=~` which is basicly will perform string matching, 
we know that the domain name would be `DOMAIN_NAME` so `==` will do.
Also if you are going to use single brackets `[` you can specify `-o` instead of `||`.

Reply:
> > if you prefer a different way of removing the empty spaces, I'm all ears ￼
> 
> From what I saw It will work w/o removing the extra line

Removing the extra line won't work with `==`. it works with the `=~` since it's a substring, but won't work with `==`. I'd much prefer to be concise and compare against the matching name instead of using a weak comparison with `=~`, And I can't get the other suggestion with `-o` to work.